src/Build/Logging/BinaryLogger/BuildEventArgsReader.cs

```diff
@@ -381,7 +398,8 @@ private BuildEventArgs ReadBuildFinishedEventArgs()
         private BuildEventArgs ReadProjectEvaluationStartedEventArgs()
         {
             var fields = ReadBuildEventArgsFields();
-            var projectFile = ReadDeduplicatedString();
+            // Null message arg is not expected
+            var projectFile = ReadDeduplicatedString() ?? string.Empty;
```

Comment:
You say null isn't expected here, but didn't you annotate it as explicitly nullable later in this PR? If it would throw an error if it's null, should we instead harden against that?

Reply:
I think I should take at least half the blame for missing the "Evaluation" part and just reading ProjectStartedEventArgs here 😆  Makes more sense now, thanks! 👍